### ISSUE TITLE
pages in sub-directory fail to render

### ISSUE THREAD

**Issue by brunoamaral** [gohugoio/hugo#4612]
The image shows the content structure, where the file messages/index.html does not render when using `hugo server` or building the full site.

If the file messages/index.html is renamed, the server picks up the change and does build the corresponding page.

<img width="260" alt="screen shot 2018-04-13 at 12 25 03" src="https://user-images.githubusercontent.com/543344/38732447-153de3d0-3f16-11e8-868d-7568eb6a930c.png">

**Comment by bep** [comment]
Thanks for getting in touch, but it feels like this is a question  that would be better suited to the [Hugo Discussion Forum](https://discourse.gohugo.io/).  We prefer to use GitHub issues only for bugs and enhancements. Feel free to update this issue with a link to the re-posted question (so that other people can find it) or add some more details if you feel this is a genuine bug. You may also find the answers you search for in [Hugo Documentation](http://gohugo.io/overview/introduction/).

**Comment by brunoamaral** [comment]
The full content tree is this: /content/stories/crypto/2018-02-15-crypto_09/messages.

Let me know if there is anything I can do to help.

**Comment by bep** [comment]
>Let meticknow if there is anything I can do to help.

You can start by reading my post and tell me if that makes sense to you. If you want to have fully static content, you can move it to /static.

**Comment by brunoamaral** [comment]
I'm still a bit confused, if that is the case shouldn't a file called messages.html be accessible at crypto_09/messages.html ? What happens is that the html file is ignored. 

Putting that folder in /static would alienate a folder from it's context and I am trying to avoid that.

**Comment by aethr** [comment]
I'm new to hugo but I was having a similar issue. My `content` folder looked like

```
content/
├── biography.md
├── index.md
└── work
    ├── 1.md
    └── _index.md
```

Running `hugo` was not picking up content in the `work` folder, but `hugo server` would correctly detect and build these pages when they changed. The fix was to rename my root `index.md` to `_index.md`.

There is a significant difference between how hugo treats `index` and `_index`.

Putting `index.html` into a folder tells hugo that this is the last piece of content in this part of the content tree and not to look further. Everything else within the subfolder is part of a "bundle" now.

Putting `_index.html` into a folder is a way of defining content and front matter that gets displayed at the root of a section.

Long story short, try renaming `index.html` to `_index.html` and see if the rest of the content in that section starts getting built.

**Comment by paraqles** [comment]
I have an similar issue. In the __about__ the file is only rendered when it is `_index.md` named.
The same behaviour is happening in the *awesome* project directory.

<pre>
<font color="#9CD9F0"><b>content</b></font>
├── _index.md
├── <font color="#9CD9F0"><b>de</b></font>
│   ├── _index.md
│   ├── <font color="#9CD9F0"><b>about</b></font>
│   │   └── index.md
│   ├── <font color="#9CD9F0"><b>projects</b></font>
│   │   ├── _index.md
│   │   └── <font color="#9CD9F0"><b>awesome</b></font>
│   │       └── index.md
│   ├── <font color="#9CD9F0"><b>team</b></font>
│   │   └── _index.md
│   └── <font color="#9CD9F0"><b>theses</b></font>
│       └── _index.md
└── <font color="#9CD9F0"><b>en</b></font>
    ├── <font color="#9CD9F0"><b>about</b></font>
    │   └── index.md
    ├── <font color="#9CD9F0"><b>projects</b></font>
    │   ├── _index.md
    │   └── <font color="#9CD9F0"><b>ourpuppet</b></font>
    │       └── index.md
    ├── <font color="#9CD9F0"><b>team</b></font>
    │   └── _index.md
    └── <font color="#9CD9F0"><b>theses</b></font>
        └── _index.md
</pre>

**Comment by aethr** [comment]
Hi @schwarmco , you might find more information about how cats-info.md is handled in the [Page Bundles](https://gohugo.io/content-management/page-bundles/) section of the documentation.

Basically, if you have `index.md` in the folder, this is the deepest file that Hugo will process and try to generate markup for. All other files / folders that are siblings of `index.md` because **resources** that belong to the section `index.md` creates. So in the `cats-info.md` example, this is a file that can be accessed / referenced from within `index.md` or its template.

If you want Hugo to also generate markup directly for `cats-info.md` then you need to rename `index.md` to `_index.md`.

Again, I'm fairly new to Hugo, but this is how it seems to work from my experience.

**Comment by hollowaykeanho** [comment]
@aethr +1 with your sharp observation. I had been investigating this issue for weeks already.
